enhancements/microshift/enabling-user-specified-featuregates.md

```diff
@@ -78,29 +87,25 @@ This approach ensures that users can experiment with feature gate capabilities w
 3. Administrator updates `/etc/microshift/config.yaml` with the chosen configuration
 4. Administrator restarts MicroShift service
 5. MicroShift detects the custom FeatureGate configuration.
-6. MicroShift writes a sentinel file to `/var/lib/microshift/`, containing the feature gate config.
+6. MicroShift writes feature gate config to `/var/lib/microshift/no-upgrade`, and logs the event
 7. The kube-apiserver propagates the feature gates to other Kubernetes components (kubelet, kube-controller-manager, kube-scheduler)
 8. Each component processes the featureGates and enables/disables the features it supports according to the configured state
 
 ##### Attempt to Revert Custom Feature Gates
 1. Administrator decides to revert custom feature gates (e.g., wants to return to default settings)
 2. Administrator modifies `/etc/microshift/config.yaml` to remove or change feature gate configuration
 3. Administrator restarts MicroShift service
-4. MicroShift detects the sentinel file exists at `/var/lib/microshift/` containing previous custom feature gates
-5. MicroShift overrides the configuration file settings with those stored in the sentinel file
-6. MicroShift logs a warning that custom feature gates cannot be reverted once applied
-7. The cluster continues to run with the original custom feature gates despite the configuration change attempt
-
-##### Attempt to Upgrade Cluster with Custom Feature Gates
-1. Administrator attempts to upgrade MicroShift to a new version (e.g., via RPM upgrade)
-2. MicroShift upgrade process checks for the existence of the sentinel file at `/var/lib/microshift/`
-3. If sentinel file exists (indicating custom feature gates are configured):
-   - The upgrade process detects the cluster is marked as non-upgradeable
-   - Upgrade is blocked with an error message indicating custom feature gates prevent upgrades
-4. Upgrade fails to proceed, preserving the current MicroShift version
+4. MicroShift detects `/var/lib/microshift/no-upgrade` with differring feature gate config.
+5. MicroShift logs a fatal error that custom feature gates cannot be reverted or changed once applied
```

Comment:
Can we allow for changing to another not upgradable FG set? Like freely change between `TechPreviewNoUpgrade`, `DevPreviewNoUpgrade`, and `customNoUpgrade` but prevent from resetting them to upgradable setting? Or this is also prohibited by OpenShift?

Reply:
We could, but we'd be diverging from OpenShift's featureset rules, which I don't think we'd want.